docs/source/internals.rst

```diff
@@ -128,3 +137,45 @@ Some rules are able to *fix* the problems they find. If this is the case,
 the rule will return a list of fixes, which describe changes to be made to
 the tree. This can include edits, inserts, or deletions. Once the fixes
 have been applied, the updated tree is written to the original file.
+
+
+.. _reflowinternals:
+
+Reflow Internals
+----------------
+
+Many rules supported by SQLFluff involve the spacing and layout of different
+elements, either to enforce a particular layout or just to add or remove
+code elements in a way sensitive to the exiting layout configuration. The
+way this is achieved is through a some centralised utilities in the
+`sqlfluff.utils.reflow` module.
+
+This module aims to achieve several things:
+* Less code duplication by implementing reflow logic in only one place.
+
+* Provide a streamline interface for rules to easily utilise reflow logic.
+
+  * Given this requirement, it's important that reflow utilities work
+    within the existing framework for applying fixes to potentially
+    templated code. We achieve this by returning `LintFix` objects which
+    can then be returned by each rule wanting to use this logic.
+
+* Provide a consistent way of *configuring* layout requirements. For more
+  details on configuration see :ref:`layoutconfig`.
+
+To support this, the module provides a :code:`ReflowSequence` class which
+allows access to all of the relevant operations which can be used to
+re-layout sections of code, or even a whole file. Unless there is a very
```

Comment:
Would this be reasonable? Partly thinking (IIUC) that we're going to name the command `sqlfluff format`.

```suggestion
reformat sections of code, or even a whole file. Unless there is a very
```

Reply:
Yeah, I was trying to not foreshadow that too much - but you're probably right.